specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,551 @@
+
+int main()
+{
+    AppNotificationActivationInfo activationInfo{ winrt::guid(c_comServerId) };
+    AppNotificationManager::Default().RegisterActivator(activationInfo);
+
+    auto args = AppInstance::GetCurrent().GetActivatedEventArgs();
+    auto kind = args.Kind();
+
+    if (kind == ExtendedActivationKind::Launch)
+    {
+        // App is launched in Foreground. So intercept toast activators via Foreground event
+        const auto token = AppNotificationManager::Default().AppNotificationActivated([](const auto&, const AppNotificationActivatedEventArgs& notificationActivatedEventArgs)
+        {
+            ProcessNotificationArgs(notificationActivatedEventArgs);
+        });
+
+        // App does Foreground Stuff Here
+
+        // Cleanup
+        AppNotificationManager::Default().AppNotificationActivated(token);
+    }
+    else if (kind == ExtendedActivationKind::AppNotification)
+    {
+        auto notificationActivatedEventArgs = args.Data().as<AppNotificationActivatedEventArgs>();
+        ProcessNotificationArgs(notificationActivatedEventArgs);
+    }
+
+    return 0;
+}
+```
+
+## Unpackaged app Registering and handling a notification
+
+Similar to packaged apps, unpackaged apps can also Register to send app notifications and act as
+targets for activation events. There is one basic difference in the Registration for
+Unpackaged apps versus their packaged counterparts: they will need to specify their own friendly App
+DisplayName and IconUri since they don't have an app manifest file that contains the same.
+
+```cpp
+int main()
+{
+    PCWSTR const c_iconName = L"icon.png";
+    std::filesystem::path binaryPath = wil::GetModuleFileNameW<std::wstring>(nullptr);
+    std::filesystem::path directoryPath = binaryPath.parent_path();
+    std::filesystem::path iconPath = directoryPath / std::filesystem::path(c_iconName);
+
+    Uri iconUri(iconPath.wstring().c_str());
+    AppNotificationActivationInfo activationInfo(L"SampleApp", iconUri);
+    AppNotificationManager::Default().RegisterActivator(activationInfo);
+
+    auto args = AppInstance::GetCurrent().GetActivatedEventArgs();
+    auto kind = args.Kind();
+
+    if (kind == ExtendedActivationKind::Launch)
+    {
+        // App is launched in FG. So intercept toast activators via FG event
+        const auto token = AppNotificationManager::Default().AppNotificationActivated([](const auto&, const AppNotificationActivatedEventArgs& notificationActivatedEventArgs)
+            {
+                ProcessNotificationArgs(notificationActivatedEventArgs);
+            });
+
+        // App does Foreground Stuff Here
+
+        // Cleanup
+        AppNotificationManager::Default().AppNotificationActivated(token);
+    }
+    else if (kind == ExtendedActivationKind::AppNotification)
+    {
+        auto notificationActivatedEventArgs = args.Data().as<AppNotificationActivatedEventArgs>();
```

Comment:
>Data [](http://example.com/codeflow?start=51&length=4)

Why does this specific case come in through the args instead of through the AppNotificationActivated event?

Reply:
See discussion earlier.